components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceOrganizationProviderServiceImpl.java

```diff
@@ -0,0 +1,331 @@
+package org.wso2.carbon.device.mgt.core.service;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.common.*;
+import org.wso2.carbon.device.mgt.core.dao.DeviceManagementDAOException;
+import org.wso2.carbon.device.mgt.core.dao.DeviceManagementDAOFactory;
+import org.wso2.carbon.device.mgt.core.dao.DeviceOrganizationDAO;
+import org.wso2.carbon.device.mgt.core.dao.DeviceOrganizationDAOException;
+import org.wso2.carbon.device.mgt.core.dao.impl.DeviceOrganizationDAOImpl;
+
+import java.util.*;
+
+public class DeviceOrganizationProviderServiceImpl implements DeviceOrganizationProviderService {
+
+    private static final Log log = LogFactory.getLog(DeviceOrganizationProviderServiceImpl.class);
+
+    DeviceOrganizationDAOImpl deviceOrganizationDAOimpl = new DeviceOrganizationDAOImpl();
+
+
+    @Override
+    public boolean addDeviceOrganization(String deviceId, String deviceName, String path, int pingMins, int state,
+                                         int isGateway) throws DeviceOrganizationException {
+        boolean isSuccess = false;
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            isSuccess = deviceOrganizationDAOimpl.addDeviceOrganization(deviceId, deviceName, path, pingMins, state,
+                    isGateway);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            log.error(e.getMessage());
+            throw new DeviceOrganizationException(e);
+        } catch (TransactionManagementException e) {
+            String msg = "Error occurred while initiating transaction";
+            log.error(msg, e);
+        } finally {
+            DeviceManagementDAOFactory.closeConnection();
+            return isSuccess;
+        }
+    }
+
+    @Override
+    public int getDeviceOrganizationStateById(String deviceId) throws DeviceOrganizationException {
+        int tempState = -1;
+        try {
+            DeviceManagementDAOFactory.beginTransaction();
+            tempState = deviceOrganizationDAOimpl.getDeviceOrganizationStateById(deviceId);
+            DeviceManagementDAOFactory.commitTransaction();
+        } catch (DeviceOrganizationDAOException e) {
+            DeviceManagementDAOFactory.rollbackTransaction();
+            String msg = "Error while getting state of device with ID: " + deviceId + "'";
+            log.error(msg, e);
+            throw new DeviceOrganizationException(msg,e);
+        } catch (TransactionManagementException e) {
+            String msg = "Error occurred while initiating transaction";
```

Comment:
no need to create a variable here.

Reply:
Fixed 0265f3eb131d37df3adb22bbfb0d6591b262c1fd